src/main/kotlin/racingcar/MovingJudgeRule.kt

```diff
@@ -2,15 +2,15 @@ package racingcar
 
 enum class MovingJudgeRule(
     private val expression: (Int) -> (Boolean),
-    private val strategy: MovingStrategy
+    val strategy: MovingStrategy
```

Comment:
[3단계 미션에서 코멘트](https://github.com/next-step/kotlin-racingcar/pull/845#discussion_r1024150415) 드렸었는데 반영이 안된 것 같아요. 🥲 

먼저 `MovingJudgeRule` 객체의 상태 값들의 네이밍이 어떤 역할을 하는지 감이 안잡히는 것 같아요. 🤔
역할을 한눈에 파악할 수 있는 네이밍을 고려해서 변경해보면 좋을 것 같아요. 😉 

그리고 `MovingJudgeRule`와 `MovingStrategy`는 강한 결합을 가지고 있어서 새로운 `MovingJudgeRule`이 만들어지면 자연스럽게 새로운 MovingStrategy도 생성되게 될거 같아요. 🤔  이 두 객체는 서로 분리하는 것 보다는 MovingStrategy가 가지는 프로퍼티를 그대로 MovingJudgeRule이 (MovingStrategy 객체 대신에) 가지도록 두는 것이 좋을 것 같아요. 


```suggestion
    val strategy: (Int) -> (Int)
```